allennlp/modules/text_field_embedders/basic_text_field_embedder.py

```diff
@@ -93,8 +94,18 @@ def forward(self, text_field_input: Dict[str, torch.Tensor], num_wrapping_dims:
         for key in keys:
             # If we pre-specified a mapping explictly, use that.
             if self._embedder_to_indexer_map is not None:
-                tensors = [text_field_input[indexer_key] for
-                           indexer_key in self._embedder_to_indexer_map[key]]
+                indexer_map = self._embedder_to_indexer_map[key]
+                if isinstance(indexer_map, collections.abc.Sequence):
+                    # If `indexer_key` is None, we map it to `None`.
+                    tensors = [(text_field_input[indexer_key] if indexer_key is not None else None)
+                               for indexer_key in indexer_map]
+                elif isinstance(indexer_map, collections.abc.Mapping):
+                    tensors = {
+                            name: text_field_input[argument]
+                            for name, argument in indexer_map.items()
+                    }
+                else:
+                    raise NotImplementedError
```

Comment:
several things on this part:

1. you should change the signature of the method to indicate that embedder_to_indexer_map can have either a list or a dict
2. I would probably just use `dict` and `list` rather than the abstract types, that's what we do everywhere else in the code 
3. I would probably move things around so that the `token_vectors = ...` is up here, that way you only have to do one `isinstance` check